samtranslator/model/api/api_generator.py

```diff
@@ -566,22 +568,32 @@ def _construct_record_sets_for_domain(
         self, domain: Dict[str, Any], api_domain_name: str, route53: Any
     ) -> List[Dict[str, Any]]:
         recordset_list = []
-        recordset = {}
 
+        recordset = self._construct_route53_routing_policy_properties(route53, {})
         recordset["Name"] = domain.get("DomainName")
         recordset["Type"] = "A"
         recordset["AliasTarget"] = self._construct_alias_target(domain, api_domain_name, route53)
         recordset_list.extend([recordset])
 
-        recordset_ipv6 = {}
         if route53.get("IpV6") is not None and route53.get("IpV6") is True:
+            recordset_ipv6 = self._construct_route53_routing_policy_properties(route53, {})
             recordset_ipv6["Name"] = domain.get("DomainName")
             recordset_ipv6["Type"] = "AAAA"
             recordset_ipv6["AliasTarget"] = self._construct_alias_target(domain, api_domain_name, route53)
             recordset_list.extend([recordset_ipv6])
 
         return recordset_list
 
+    def _construct_route53_routing_policy_properties(
+        self, route53_config: Dict[str, Any], recordSet: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        if route53_config.get("RoutingPolicy") is not None and route53_config.get("RoutingPolicy") == "Latency":
+            region = boto3.session.Session().region_name
+            recordSet["SetIdentifier"] = region
+            recordSet["Region"] = region
```

Comment:
Thoughts on simply adding `SetIdentifier` and `Region` instead? It's not clear to me why setting those enables latency-based routing, and it might turn out to be too much maintenance in the future, if for whatever reason a different combination is desired.

Reply:
This was my first pass, but I decided to change it for a couple of reasons:

 - The other properties (i.e. HostedZoneName) in Route53 config are technically for the Route53::RecordSetGroup, where these properties are for the Route53::RecordSet.. is it confusing to combine these properties into the same entity? If not, this may be the way to go - see below.
 
- According to [this documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset-1.html#:~:text=%3A%20No%20interruption-,Region,-Latency%2Dbased%20resource), `Region` is only valid for Latency-based routing policies. And [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset-1.html#:~:text=%3A%20No%20interruption-,SetIdentifier,-Resource%20record%20sets) you'll see that the `SetIdentifier` is provided for any policy type other than `Simple`. This allows us to add additional routing policies without requiring the user to look up what in particular is required for a given policy. 

However, it requires us to lock down the region, which some users may not like. In addition, it becomes more complicated with the other policies (think GeoIp). 

If you're ok with combining the RecordSetGroup and RecordSet properties, it covers everything - happy to refactor it as such.

Reply:
That said, going the other route introduces an interesting predicament as well - setting `Region` and `GeoLocation` together is invalid (and is true for every policy-specific property) - should we be validating at this level as well?